### PR TITLE
Allow admin/moderator to disable user from inviting

### PR DESCRIPTION
**Pull request by nyanpasu**
```
2016-02-11 17:06:05	nekomune	hi. i was banned yesterday because of
the actions of one of my invitees, yui... can i discuss it here?

2016-02-11 17:07:24	nekomune	i understand that i'm responsible for
who i'm invited, and i have told him multiple times to be careful with his
words and contentious attitude.

2016-02-11 17:08:50	nekomune	i think a full ban for me is unfair. ¬ ¬

2016-02-11 20:54:48	asthasr|away	nekomune: It's one of the policies of
the site, an attempt to keep it self-policing so that people don't invite
recklessly. You might message jcs about it (he is the admin).

2016-02-11 20:55:34	nekomune	cheers.

2016-02-11 20:56:34	asthasr	nekomune: As a sign of good faith you might
submit a pull request for removing someone's invite privileges :p
```

:p

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -37,6 +39,32 @@ def invite
...
+    target = User.where(:username => params[:username]).first
+    if !target
+      flash[:error] = "Invalid user."
+      return redirect_to "/"
+    end
+
+    target.disable_invite_by_user_for_reason!(@user, params[:reason])
+
+    flash[:success] = "User has had invite capability disabled."
+    return redirect_to user_path(:user => target.username)
```
Comment (pushcx) [inline app/controllers/users_controller.rb]: The `return` here and on line 65 don't do anything

Comment (nyanpasu) [inline app/controllers/users_controller.rb]: Noted, changed.

File: app/models/user.rb [hunk 2]
```diff
@@ -108,6 +110,25 @@ def average_karma
     end
   end
 
+  def disable_invite_by_user_for_reason!(disabler, reason)
+    self.disabled_invite_at = Time.now
+    self.disabled_invite_by_user_id = disabler.id
+    self.disabled_invite_reason = reason
+
+    self.delete!
```
Comment (jcs) [inline app/models/user.rb]: wat

**Comment by jcs** [comment]
The mailer is probably overkill, just sending a private message would suffice.

I hate how many settings columns there are already, and this adds three more, but I guess it will be temporary.  #245

**Comment by jcs** [comment]
@nyanpasu your account is unbanned

**Pull request merged**